ballerina/modules/bulk/client.bal

```diff
@@ -48,10 +48,10 @@ public isolated client class Client {
         self.clientConfig = auth.cloneReadOnly();
 
         http:ClientOAuth2Handler|http:ClientBearerTokenAuthHandler|error httpHandlerResult;
-        if self.clientConfig is http:OAuth2RefreshTokenGrantConfig {
-            httpHandlerResult = trap new (<http:OAuth2RefreshTokenGrantConfig>self.clientConfig);
+        if auth is http:OAuth2RefreshTokenGrantConfig {
+            httpHandlerResult = trap new (<http:OAuth2RefreshTokenGrantConfig>auth);
```

Comment:
Can't we remove the cast expressions as the type is already checked within the if statement?
```suggestion
            httpHandlerResult = trap new (auth);
```

Reply:
Check all the applicable places

Reply:
Updated in both bulk and soap modules